lib/Dialect/FIRRTL/Transforms/CheckCombLoops.cpp

```diff
@@ -248,12 +248,12 @@ class DiscoverLoops {
   void preprocess(SmallVector<Value> &worklist) {
     // All the input ports are added to the worklist.
     for (BlockArgument arg : module.getArguments()) {
-      auto argType = arg.getType();
+      auto argType = cast<FIRRTLType>(arg.getType());
       if (argType.isa<RefType>())
         continue;
```

Comment:
Note to self: follow-up to get this working through references (cc #4691), thanks for getting this working for non-base types!